controllers/apps/component_plan_builder.go

```diff
@@ -222,7 +223,7 @@ func (c *componentPlanBuilder) reconcileDeleteObject(ctx context.Context, vertex
 	// The additional removal of DBClusterFinalizerName in the component controller is to backward compatibility.
 	// In versions prior to 0.9.0, the component object's finalizers includes DBClusterFinalizerName.
 	// Therefore, it is necessary to remove DBClusterFinalizerName when the component is scaled-in independently.
-	finalizers := []string{constant.DBComponentFinalizerName, constant.DBClusterFinalizerName}
+	finalizers := []string{constant.DBComponentFinalizerName, constant.DBClusterFinalizerName, rsmcore.FinalizerName}
```

Comment:
better to add a comment explaining why the RSM finalizer needs to be removed from the component controller

Reply:
fixed [e1e6052](https://github.com/apecloud/kubeblocks/pull/7087/commits/e1e6052ca89391e27ea523370114bb7e3d780cbc)